lib/controller/mixins/next-step.js

```diff
@@ -82,6 +82,10 @@ module.exports = Controller => class extends Controller {
             });
         }
 
+        if (typeof nextStep === 'function') {
+            nextStep = nextStep(req, res);
```

Comment:
perhaps also change `if (typeof condition === 'string') {` to `if (typeof condition === 'string' || typeof condition === 'function') {` on line 66 to allow defaults in sets to also be functions and not having to do the fn true trick